### PR TITLE
Fixbug: give the highest priority for the system variables

### PR DESCRIPTION
**Pull request by AlekseyLeshko**
Test case
Have `.defaults`, `.example` files.
If I try set system env variable in CI, I got value from `.deafults` file

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -64,7 +64,11 @@ class Dotenv {
 
     // add the leftovers
     if (safe) {
-      Object.assign(vars, env)
+      Object.keys(env).forEach((key) => {
```
[the remark this answers is not in the document]
Comment (AlekseyLeshko) [reply src/index.js]: 'Object.assign' does not approach. We need another method for a merge of objects. 
'Object.assign' replace all variables from `env` in `vars`.
System env variables were being replaced, it's suddenly and wrong behavior.

Set default variables for a fast start in the project.
Set example to check the required variables.
Set system env variables, for example for control of build mode(development, production and etc)
And now `Object.assign` replace variable for build mode

I can create a small repository for demo.

[the remark this answers is not in the document]
Comment (marcofugaro) [reply src/index.js:67]: Is this blocked?

**Comment by mrsteele** [comment]
@AlekseyLeshko Could you rebase with the new code? We replaced mocha/chai for jest so you might have to just copy those tests over using the new framework API.

**Pull request merged**